src/sentry/sentry_apps/api/bases/sentryapps.py

```diff
@@ -373,6 +380,9 @@ def has_object_permission(self, request: Request, view, installation):
         if superuser_has_permission(request):
             return True
 
+        assert isinstance(
+            request.user, (RpcUser, User)
+        ), "user must be authenticated to check if they're a sentry app"
```

Comment:
We should be able to start using `user.is_authenticated` to discriminate the anonymous user part of the union. See https://github.com/getsentry/sentry/pull/78057#discussion_r1773990542